src/lib/update-lock.ts

```diff
@@ -94,18 +122,22 @@ export const readLock: LockFn = Bluebird.promisify(locker.async.readLock, {
 function dispose(release: () => void): Bluebird<void> {
 	return Bluebird.map(_.keys(locksTaken), (lockName) => {
 		delete locksTaken[lockName];
-		return lockFile.unlockAsync(lockName);
+		return lockfile.unlock(lockName);
 	})
 		.finally(release)
 		.return();
 }
 
-const lockExistsErrHandler = (err: Error, release: () => void) => {
-	let errMsg = err.message;
-	if (EEXIST(err)) {
+const lockfileErrHandler = (err: Error, release: () => void) => {
+	let error: any;
+
+	// If error code is 73, updates.lock file already exists.
+	if (LockfileExistsError(err as ChildProcessError)) {
```

Comment:
If the conversion to `LockFileExistsError` happens inside the `lockfile` module then this just needs to do

```
if (err instanceof LockFileExistsError)
```

Reply:
Actually you don't need to check the error type, any error should cause the update-lock function to throw.

Reply:
For convenience when building a Supervisor-specific lockfile exists error message, checking may still help to avoid some regex `.match` calls:
```
const lockfileErrHandler = (err: Error, release: () => void) => {
	let error = err;

	if (err instanceof lockfile.LockfileExistsError) {
		let errMsg = (err as lockfile.LockfileExistsError).stderr;

		// Extract appId|appUuid and serviceName from lockfile path for log message
		// appId: [0-9]{7}, appUuid: [0-9a-f]{32}, short appUuid: [0-9a-f]{7}
		const pathMatch = errMsg.match(
			/\/([0-9]{7}|[0-9a-f]{32}|[0-9a-f]{7})\/(.*)\/(?:resin-)?updates.lock/,
		);
		if (pathMatch && pathMatch.length === 3) {
			errMsg = `Lockfile exists for ${JSON.stringify({
				serviceName: pathMatch[2],
				[/^[0-9]{7}$/.test(pathMatch[1]) ? 'appId' : 'appUuid']: pathMatch[1],
			})}`;
		}

		error = new UpdatesLockedError(errMsg);
	}

	return dispose(release).throw(error);
};
```

Reply:
I think we need to move away from this, we can discuss tomorrow in a call if you like. Parsing the error message is a not a good practice imho.

Reply:
From this I mean, from parsing the error message :p

Reply:
The main upside to adding error message parsing to extract the serviceName and appId/uuid is for the clarity of the message which I rather liked.. we could move away from parsing but I'd still want the exact source of the error communicated clearly, so it would mean surfacing a file path directly in the error message. I like surfacing the lockfile path directly because it tells the user exactly where to remove the lockfile, but dislike it because it's overly verbose (the path name is too long a block of letters together, leading to less than ideal UX).

Reply:
I do not know how we can do the following things simultaneously:
- Remove the lockfile error handler from update-lock while still letting users know the exact source of the lockfile error
- Throw a generic error in the lockfile module that doesn't reference serviceName or appId in any way, but still communicate this with good UX to the user, without sending a long file path

Reply:
No longer parsing the error message, we're just checking if the thrown error is an instance of LockfileExistsError or not.